gcp_variant_transforms/testing/integration/run_tests_common.py

```diff
@@ -40,43 +54,69 @@ class TestCaseFailure(Exception):
 class TestRunner(object):
   """Runs the tests using pipelines API."""
 
-  def __init__(self, tests):
-    # type: (List[Union[PreprocessorTestCase, VcfToBQTestCase]]) -> None
+  def __init__(self, tests, revalidate=False):
+    # type: (List[List[TestCaseInterface]], bool) -> None
+    """Initializes the TestRunner.
+
+    Args:
+      tests: All test cases.
+      revalidate: If True, only run the result validation part of the tests.
+    """
     self._tests = tests
     self._service = discovery.build(
         'genomics',
         'v1alpha2',
         credentials=GoogleCredentials.get_application_default())
-    self._operation_names = []  # type: List[str]
-    self._responses = []  # List[Dict]
+    self._revalidate = revalidate
+    self._operation_names_to_test_states = {}  # type: Dict[str, TestCaseState]
 
   def run(self):
     """Runs all tests."""
-    for test in self._tests:
-      # The following pylint hint is needed because `pipelines` is a method that
-      # is dynamically added to the returned `service` object above. See
-      # `googleapiclient.discovery.Resource._set_service_methods`.
-      # pylint: disable=no-member
-      request = self._service.pipelines().run(body=test.pipeline_api_request)
-      operation_name = request.execute()['name']
-      self._operation_names.append(operation_name)
-    self._wait_for_all_operations_done()
+    if self._revalidate:
+      for test_cases in self._tests:
+        for test_case in test_cases:
```

Comment:
As discussed offline: I think we should just validate the last test in this case.

Reply:
Thanks very much on catching this issue! Done now. PTAL.